modules/openapi-generator/src/main/resources/C-libcurl/apiKey.c.mustache

```diff
@@ -9,6 +9,12 @@ keyValuePair_t *keyValuePair_create(char *key, void *value) {
     return keyValuePair;
 }
 
+keyValuePair_t* keyValuePair_create_allocate(char* key, double value) {
```

Comment:
Hi @michelealbano, thank you for your great work. I have some questions here.

1. Is this function "keyValuePair_create_allocate" used now?

2. If the data type of value is not "double", this function cannot be used.

3. Form my view, user should manage(allocate and free) the memory.  If we (OpenAPI-Generator) provide the allocating function, we should also provide the free function.

Reply:
It is to be used in the main.
Currently, the generated code for models says:
`if(cJSON_AddNumberToObject(localMapObject, localKeyValue->key, *(double *)localKeyValue->value) == NULL)`
thus it considers the value as a pointer. The keyValuePair_create limited itself to copying the value. Imagine I wanted to have a ("numberOfTeeth", 5), I had to do this in my main:
```
int number = 5; // since I need a lvalue
double* val1 = malloc(sizeof(number));
memcpy(val1, &number, sizeof(number));
keyValuePair_create("numberOfTeeth", val1);
```

I tried to put directly the number (5) in the value, casting int to void *, but I would have to change a great deal of generated code for models, for example the ones I showed at the beginning of this message.

We are actually providing free functions! For example, I have this in my model.c:
```
void orchestration_form_request_dto_free(orchestration_form_request_dto_t *orchestration_form_request_dto) {
    listEntry_t *listEntry;
    list_ForEach(listEntry, orchestration_form_request_dto->orchestration_flags) {
        keyValuePair_t *localKeyValue = (keyValuePair_t*) listEntry->data;
        free (localKeyValue->key);
        free (localKeyValue->value);
    }
    list_free(orchestration_form_request_dto->orchestration_flags);
    free(orchestration_form_request_dto);
}
```

Reply:
OK. I got it. No problems.